### ISSUE TITLE
Upload never completes with slow connections

### ISSUE THREAD

**Issue by jidanni** [commons-app/apps-android-commons#3972]
Please test uploading with a 2 megabyte image, over a household ADSL 2M/64K connection.

Yes, that is right, 64 KB upload speed.

You will see the upload never completes.

**Comment by nicolas-raoul** [comment]
I also have a super-slow connection sometimes.
Does it happen when uploading a single image, or only for sets of several images?
Are you using the alpha version? If not, can you try it? And use Settings>Send log when that happens.
Thanks!

**Comment by jidanni** [comment]
Yes, single image. Note it may be beyond your control, and actually a
https://phabricator.wikimedia.org/T205619
issue.
All I know about is #3976, and no "alpha".

**Comment by misaochan** [comment]
Just to clarify, @jidanni , this fails for you on the Upload Wizard as well?

**Comment by jidanni** [comment]
> Just to clarify, @jidanni , this fails for you on the Upload Wizard as well?

Correct. Which means for half the month (after I use up my 4G LTE (high speed) cell plan allowance) there is no way to upload images (even just 2MB ones.) As my home ADSL is just 64K upload speed, and any convenience store Wi-Fi is too far away back in town.

**Comment by misaochan** [comment]
Hmmm, OK. That is very annoying, but unfortunately we can't do anything about it because we only handle the Android app, for the Commons server we depend on WMF. I will tag this as upstream instead then.
